### ISSUE TITLE
Can InvisibleButton allow size 0?

### ISSUE THREAD

**Issue by alien-brother** [ocornut/imgui#7623]
### Version/Branch of Dear ImGui:

Version 1.90.5, Branch: master

### Back-ends:

Any

### Compiler, OS:

Any

### Full config/build information:

Any

### Details:

**My Issue/Question:**

Currently, there's a comment and an assertion in `ImGui::InvisibleButton()`:
```cpp
    // Cannot use zero-size for InvisibleButton(). Unlike Button() there is not way to fallback using the label size.
    IM_ASSERT(size_arg.x != 0.0f && size_arg.y != 0.0f);
```
Would it be possible to just create a button of size (0, 0) in this case? In particular, `Dummy()` does allow size (0, 0), meaning it's probably not probably an issue for all the infrastructure behind `ItemAdd()`?
Otherwise, if you use `InvisibleButton()` to create a custom button-like widget, this assertion becomes easy to hit: set frame padding to 0 and give the button an empty label. Admittedly, zero-size buttons aren't particularly pretty.

### Minimal, Complete and Verifiable Example code:

```cpp
ImGui::InvisibleButton("##button", ImVec2(0.0f, 0.0f));
```

**Comment by ocornut** [comment]
It's not possible what's what the assert is saying. I don't understand why you would want it?

**Comment by ocornut** [comment]
To clarify

**Comment by ocornut** [comment]
To clarify, 
- your custom widget can add an extra if statement to avoid submitting empty one or handle this edge case.
- whereas if we remove this assert and claim to support it, users may accidentally have issues passing a zero value and never understanding where the widget is gone. Especially as for some API (0,0) may mean "fill available space" but not for this one.

**Comment by ocornut** [comment]
I am assuming you can add the extra if in your custom widget, let us know if you feel this is a problem.
Closing.

**Comment by alien-brother** [comment]
Obviously, a custom widget can make an extra check before calling `InvisibleButton()`. I created the issue from the point of view, that in general, size 0 is valid for ImGui widgets, cf. `Dummy()`, `SmallButton()`, and then there's one widget that crashes the whole program when given size 0.

**Comment by ocornut** [comment]
I thought about it again and would like to confirm my first impression.

`Button({0,0})` uses label size aka automatic sizing, just as many other widgets treat {0,0} as automatic sizing. `InvisibleButton()` doesn't have a label. If you pass {0,0} following a computation and the widget is automatic sized it's likely a mistake but one that becomes easy to spot. 

I don't think there's a neat parallel possible with other widgets, it's just that `InvisibleButton()` cannot support automatic sizing.

`Dummy({0,0})` indeed works but generally it wouldn't be "harmful" if you submit this or miscompute.

**Comment by ocornut** [comment]
@alien-brother I have revaluated this and changed my mind, and pushed the change 791ad9b82db44ada9fedb3e26b2d900974ac0959. 
Apologies for the previous decision and taking a while to get back to it: I omitted the fact that `BeginChild()` and `BeginTable()` already support (0,0) for fit to content even in situations where size isn't reported inside-out. While evaluating #9166 I realized `InvisibleButton()` already de-facto supported negative value to align to right/bottom edge, so there's no reason to not support (0,0) there.
